### ISSUE TITLE
Stocks transfer leads to unexpected cost values

### ISSUE THREAD

**Issue by fdw** [simonmichael/hledger#858]
In my journal file, I transferred some stocks from one account to the other:
```
assets:acct1   =0AAPL
assets:acct2
```

When I now call `hledger balancesheet --cost`, I would expect no costs in acct1, and all of them in acct2, resolved for the main currency. However, I find something like this:
```
assets:acct1 || 100€, -10AAPL
assets:acct2 || 10AAPL
```

Why is there both the cost in acct1, and also the stocks (negatively)? Why is there no actual cost in acct2? Is this intended behavior? Am I missing something else?

**Comment by simonmichael** [comment]
It's hard to follow.. could you paste more detail, ie a reproducible example ?

**Comment by simonmichael** [comment]
But a general tip: to troubleshoot unexpected balances, use the register, eg `hledger reg assets:acct1`.

**Comment by fdw** [comment]
Sorry that I wasn't clear enough. I'll try again with a longer example (please excuse that it isn't my real finances 😉):

I have several transactions buying a stock:
```
2018/06/01 Stocks
    assets:acct1    5 AAPL @@ 40
    assets:cash    -40

2018/07/01 Stocks
    assets:acct2    5 AAPL @@ 60
    assets:cash    -60

P 2018/08/01 AAPL  15
```

Now, `hledger bs -B` and `hledger bs -V` show the expected values: The cost is 100, the current value is 150.

I now add a second depot and want to transfer all my stocks there:
```
2018/08/03 Transfer
    assets:acct1   =0 AAPL
    assets:acct2
```

What I would expect for `hledger bs -B` and `hledger bs -V` is something like this:
`hledger bs -B`:
```
acct1: 0
acct2: 100
```
and `hledger bs -V`:
```
acct1: 0
acct2: 150
```

However, what I get for `hledger bs -B` is:
```
acct1: 100, -10 AAPL
acct2: 10 AAPL
```
`hledger bs -V` works fine (i.e. is 0 for acct1, 150 for acct2).

I'm not sure why the cost is given like this.

**Comment by simonmichael** [comment]
Great, now I follow. Short answer: when prices are involved, sometimes the more explicit your entries are, the better. This works as you expect:
```
2018/08/03 Transfer
    assets:acct1  -5 AAPL @@ 40 = 0 AAPL
    assets:acct2   5 AAPL @@ 40 
```

Your entry didn't specify the price of the amount being transferred, and hledger isn't smart enough to know that you probably meant the @@ 40 lot that's already in the account, instead it follows what you wrote literally, transferring an unpriced lot. Hope this helps ?

**Comment by simonmichael** [comment]
It's possible of course but can it be done in a reliable and intuitive way ? See if you can write down a specification for it. It seems a bit tricky to me.

Emacs macros are a wonderful power tool for this kind of mass edit.

**Comment by fdw** [comment]
I'm more of a vim user, but together with some shell magic, I managed ;)

Are entries always coupled with their prices? I had hoped hledger would always try to use the existing lots (and their prices) for the automatic entries, and throw an error if there is some ambiguity. Instead, it probably only looks at the amounts, but not at their prices, correct?

**Comment by fdw** [comment]
Hi, I'm sorry...

What I meant is more or less this:
> hledger isn't smart enough to know that you probably meant the @@ 40 lot that's already in the account, instead it follows what you wrote literally, transferring an unpriced lot.

I think it would suffice if hledger tried to use the lot that already exists instead of generating a new, unpriced one.
For starters, it should also be enough if this only works if the lot is unambiguous (i.e. you're transferring/selling all stocks). But of course, if I could instruct hledger to always use the oldest one, or the newest one, and it could then choose the appropriate lot(s) from that, that would be great!

**Comment by fdw** [comment]
Since my question has been answered, do you want to close the ticket or keep it open to track a possible improvement?

**Comment by simonmichael** [comment]
Thanks @fdw. I think we can close this as a support incident clarifying hledger's feature set. Currently hledger moves exactly the (possibly priced) amounts that you specify, without any deeper knowledge about whether that makes sense in the real world. A proposal describing smarter behaviour could happen on a new issue or the mail list.
